subprojects/enterprise-operations/src/main/java/org/gradle/operations/execution/ExecuteWorkBuildOperationType.java

```diff
@@ -47,6 +48,11 @@ public interface Details {
          */
         @Nullable
         String getWorkType();
+
+        /**
+         * The identity of the executed unit of work.
```

Comment:
:x: This Javadoc doesn't add anything, though we could actually use it to explain what we mean by the "identity" of work. What will it be used for and how? What uniqueness guarantees do we have for it? etc.

Reply:
Not sure, but I guess there should be some if changes had to be made to the identity of mutable transforms recently, right? What can someone expect when calling this method? Is it a unique ID for the build, the build tree, a globally unique ID, what?

As for what is it and how it's used, I guess we should document why the method is here in the first place. How will the scans plugin use this information?

Reply:
I think what this id is depends a lot on the type of work. This is the identity is the identity from the execution engine. My understanding is that it is supposed to uniquely identify a unit of work with an execution history store. For example, for tasks, this is the task path that uniquely identifies the task within the build, and there is one execution history per build (and possibly multiple per build tree).

For transforms, it depends on the type of transform:
- For non-project transforms, the execution history is Gradle user home scoped, so the id is unique per Gradle user home, especially in the build tree.
- For project transforms, the execution history is per build as well, so it needs to be unique, at least within the build. It wasn't, and that is what my PR fixed. My PR even made the id unique within the whole build tree.

For now, this will be used to identify a transform execution - the only unit of work for which we currently emit the operation - in build scans within the current build tree. If we want to identify an arbitrary unit of work, we'd also need to add an identification of the execution history to this operation. Or we'd expect this String to already contain the identification of the execution history.

Reply:
IIUC, this is a unique ID for the work among all similar types of work in the scope of a build tree, and that's what you want to use it for in the scans plugin (=identify a piece of work in a build tree). If that's correct, then this is what the Javadoc should say.

Reply:
What I am saying is that for tasks, it isn't. And I don't know yet how'd we use the identity when we use the general execution operation for tasks. It feels like the details would get a new field that identifies the execution history ("scope" of the identity) instead of folding the build path for tasks into the identity.